mdsuite/transformations/transformations.py

```diff
@@ -285,7 +285,10 @@ def _prepare_database_entry(self, species: str, system_tensor=False):
             output_length = 1
             path = join_path(self.output_property.name, self.output_property.name)
         else:
-            output_length = self.experiment.species[species]["n_particles"]
+            try:
+                output_length = self.experiment.species[species]["n_particles"]
+            except KeyError:
+                output_length = self.experiment.molecules[species]["n_particles"]
```

Comment:
why are the molecules not a scpecies? They behave mostly the same

Reply:
Yep but this is a problem I came across when initially implementing it and it can be resolved in a nicer way now that we have a data class for species I guess. The issue arises when I compute something like ionic current or any flux essentially. It collects all of the data in the species list which means that the molecules, which are just representations of groups of particles, will be included in that calculation, which then will be wrong. 

This could be avoided by having an attribute in the data class like group: true or something which tells the transformations to avoid loading these ones for these transformations. Conversely one may want to compute a flux only from the molecules in the case where atomic particle charges are used whereas the molecules contain complete charges which can be easier to use if this data is not stored in the simulation.

If you have a preference let me know.

Reply:
Not fully thought out, but could a molecule derive from species? We could do that if __all__ species attributes also exist for molecules (which I think they do??). Molecules would then only add their specific information.

The transformations/calculators by default loop over all species, we could make it like ``run_transformation(subject_names:list[str]=None, all_species = True, all_molecules = False`` so the user can switch between species, molecules or their own list of species/molecules

Reply:
So I was thinking that in the species data class we can have a:

```python
group_representation = True
```
or something. In this case, similar to what you have here, they can by default ignored in transformation and computations or includes at will. In this case, if a `species` is actually a group of particles or a representation of several particles. I like the generic use of `group` here to keep in general in case of different data, e.g polymer or just a group of colloids or whatever else.

Reply:
no objection on the general thought here from me. But I prefer it a child of species and not to add a default-inactive argument to species. In that way the normal species stays "unpolluted" by stuff it does not need for itself

Reply:
Ahh okay so make the Molecules class a child of Species and then check for it in the transformations calculations, a good idea.

Reply:
@PythonFZ  your input here would be quite helpful in my opinion.

Reply:
Can we move this to an issue? I think this needs more planing and would affect most of the calculators as well? I do like the idea of combining them but it would probably be best in a dedicated PR?

Reply:
issue #477